internal/infrastructure/worker/work.go

```diff
@@ -0,0 +1,90 @@
+package worker
+
+import (
+	"strconv"
+
+	"github.com/andrsj/go-rabbit-image/pkg/logger"
+)
+
+const (
+	level75 = 75
+	level50 = 50
+	level25 = 25
+)
+
+// Start() method of the worker struct.
+func (c *worker) Start() {
+	// Start consuming messages from the Consumer
+	messageCh, errorCh := c.client.MustConsumeMessages()
+	c.logger.Info("Consumer has started", nil)
+
+	go func() {
+		for {
+			select {
+			case message := <-messageCh:
+				// Decode the image from the message body
+				img, contentType, err := decodeImage(message.Body)
+				if err != nil {
+					c.logger.Error("Decoding image", logger.M{"error": err})
+					c.logger.Warn("Skipping image", logger.M{"image_id": message.ImageID})
+
+					continue
+				}
+
+				// Create image with 100% quality
+				go func() {
```

Comment:
You don't need a goroutine here, since this is an async worker and it could wait.
More to read - https://www.storj.io/blog/production-concurrency